src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -1597,3 +1599,37 @@ cmp arr1 arr2 =
 instance NFData (Array a) where
     {-# INLINE rnf #-}
     rnf Array {} = ()
+
+strip :: forall a m. (Storable a, MonadIO m) =>
+    (a -> Bool) -> Array a -> m (Array a)
+strip eq arr@Array{..} =
+    let p = unsafeForeignPtrToPtr aStart
+        q = aEnd
+        len = length arr
+        st = getStart p len
+        end = do
+            st' <- st
```

Comment:
The action `st` is being called at two places, we are effectively searching for start twice, instead we can call it once and use the result.